Add predefined model interface doc (#7830)
opensearch-project/documentation-website: _ml-commons-plugin/api/model-apis/register-model.md

### The `interface` parameter

The model interface provides a highly flexible way to add arbitrary metadata annotations to all local deep learning models and externally hosted models in a JSON schema syntax. This annotation initiates a validation check on the input and output fields of the model during the model's invocation. The validation check ensures that the input and output fields are in the correct format both before and after the model performs inference.

To register a model with a model interface, provide the `interface` parameter, which supports the following fields.

Field | Data type | Description                         
:---  | :--- |:------------------------------------
`input`| Object | The JSON schema for the model input. |
`output`| Object | The JSON schema for the model output. |

> The input and output fields are evaluated against the provided JSON schema. You do not need to provide both input and output fields simultaneously.

Reviewer: ```suggestion
The input and output fields are evaluated against the provided JSON schema. You do not need to provide both fields simultaneously.
```

The input and output fields will be evaluated against the separately provided JSON schema. You do not necessarily need to provide both input and output fields simultaneously.

> Specifically, for users' convenience, we have provided a set of predefined model interface that can be generated automatically during model registration, if the user choose to follow the [connector blueprints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/). 

Reviewer: ```suggestion
Specifically, for users' convenience, we have provided a set of predefined model interfaces that can be generated automatically during model registration if the user chooses to follow the [connector blueprints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/). 
```

> Specifically, for users' convenience, we have provided a set of predefined model interface that can be generated automatically during model registration, if the user choose to follow the [connector blueprints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/). 

Reviewer: "users'" => "your"?, "the user chooses" => "you choose"?

> The predefined model interface is generated based on the connector blueprint and the model's metadata, so user needs to follow the blueprint to create the connector strictly in order to avoid exceptions.

Reviewer: ```suggestion
The predefined model interface is generated based on the connector blueprint and the model's metadata, so the user must strictly follow the blueprint when creating the connector in order to avoid exceptions.
```

> The predefined model interface is generated based on the connector blueprint and the model's metadata, so user needs to follow the blueprint to create the connector strictly in order to avoid exceptions.

Reviewer: "the user" => "you"?

> The list of connector blueprints we currently support predefined model interface are:

Reviewer: ```suggestion
The following are the currently supported connector blueprints for creating predefined model interfaces:
```

> - [amazon comprehend connector blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/amazon_comprehend_connector_blueprint.md)

Reviewer: ```suggestion
- [Amazon Comprehend connector blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/amazon_comprehend_connector_blueprint.md)
```

> - [amazon textract connector blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/amazon_textract_connector_blueprint.md) (Please note that predefined model interface is only available for the `DetectDocumentText` API, `DetectEnities` API is not supported currently)

Reviewer: ```suggestion
- [Amazon Textract connector blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/amazon_textract_connector_blueprint.md) (Note that a predefined model interface is only available for the `DetectDocumentText` API; the `DetectEnities` API is not currently supported).
```
